### PR TITLE
Codemod: Fix csf-hoist-story-annotations for variable default exports

### PR DESCRIPTION
**Pull request by shilman**
Issue: #11894

## What I did

Fix edge case of variable default exports.

## How to test

See attached test case

### CODE REVIEW

**Comment by shilman** [comment]
self-merging @tmeasday

**Review by jimmyandrade — COMMENTED** [review]
LGTM

File: lib/codemod/src/transforms/__testfixtures__/csf-hoist-story-annotations/.output.snapshot [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (tmeasday) [inline lib/codemod/src/transforms/__testfixtures__/csf-hoist-story-annotations/.output.snapshot:1]: @shilman did you mean to check this in?

Comment (shilman) [reply lib/codemod/src/transforms/__testfixtures__/csf-hoist-story-annotations/.output.snapshot:1]: oops, good catch!

**Pull request merged**